google-cloud-firestore/src/main/java/com/google/cloud/firestore/QueryPartition.java

```diff
@@ -0,0 +1,61 @@
+package com.google.cloud.firestore;
+
+import javax.annotation.Nullable;
+
+/**
+ * A split point that can be used by in a query as a starting or end point for the query results.
+ * The cursors returned by {@link #getStartAt()} and {@link #getEndBefore()} can only be used in a
+ * query that matches the constraint of query that produced this partition.
+ */
+public class QueryPartition {
```

Comment:
Thoughts on changing this class name to `QueryCursorRange`?

Reply:
Yesterday I felt okay with changing this. Today I am a little more hesitant (but not strongly opposed). QueryPartition seems shorter and relates to both the API name introduced in this PR and the backend feature name. Is `QueryCursorRange` a name that follows a precedent outside of our product?

Reply:
That's fair. I don't feel strongly about it, and agree that keeping it in sync with the backend api is more important.